torch/csrc/jit/passes/shape_analysis.cpp

```diff
@@ -970,6 +968,33 @@ class ShapePropagator {
       return {};
     };
 
+    // Requirements:
+    //   dims           : 0 if dim is None, otherwise preserved if keepdim == false or 1 smaller otherwise
+    //   scalar type    : preserved
+    //   device         : preserved
+    //   tensor inputs  : 1
+    //   tensor outputs : 1
+    // Additionally:
+    //   - First input should be the only tensor input
+    //   - Has a bool keepdim argument
+    static const register_formula_for argminmax{
+        {
+            "aten::argmax(Tensor self, int? dim, bool keepdim) -> Tensor",
+            "aten::argmin(Tensor self, int? dim, bool keepdim) -> Tensor",
+        },
+        [](Node* node) -> type_vec_t {
+          if (auto type =
+                  node->input(0)->type()->cast<DimensionedTensorType>()) {
+            if (node->input(1)->type()->kind() == c10::TypeKind::NoneType) {
+              return {type->withDim(0)};
+            } else {
+              return multidim_reduce_with_postprocess(
+                  node, /*num_reduce_dim=*/1, /*integer_upcast=*/false);
```

Comment:
nit: 
  Argument name in comment (num_reduce_dim) does not match parameter name (num_reduced_dim).
  Argument name in comment (integer_upcast) does not match parameter name (upcast_integer).

Reply:
@VitalyFedyunin fixed